### PR TITLE
Fix the location that selecting a mark uses

### LINKED ISSUE #17131: [1.21] Selecting commands, output with the context menu is totally in the wrong place
open pwsh, with shell integration enabled

do a 

```
cd Z:\dev\public\OpenConsole\
git status
dir
foo
whatever
```

try and right-click select the output of `dir`, and it's... wrong:

image

surely, I'm the murderer, but I could have swore there were tests for this

### PR DESCRIPTION
**Pull request by zadjii-msft**
I think this subtly regressed in #16611. Jump to https://github.com/microsoft/terminal/commit/90b8bb7c2d8932fc601aed9c26f13c77d5a1321d#diff-f9112caf8cb75e7a48a7b84987724d754181227385fbfcc2cc09a879b1f97c12L171-L223

`Terminal::SelectNewRegion` is the only thing that uses the return value from `Terminal::_ScrollToPoints`. Before that PR, `_ScrollToPoints` was just a part of `SelectNewRegion`, and it moved the start & end coords by the `_VisibleStartIndex`, not the `_scrollOffset`.

Kinda weird there weren't any _other_ tests for `SelectNewRegion`?

I also caught a second bug while I was here - If you had a line with an exact wrap, and tried to select that like with selectOutput, we'd explode.

Closes #17131

### CODE REVIEW

File: src/cascadia/TerminalControl/ControlCore.cpp [hunk 1]
```diff
@@ -2620,12 +2620,27 @@ namespace winrt::Microsoft::Terminal::Control::implementation
...
     }
+
+    // Select the region of text between [s.start, s.end), in buffer space
     void ControlCore::_selectSpan(til::point_span s)
     {
+        // s.end is an _exclusive_ point. We need an inclusive one. But
+        // decrement in bounds wants an inclusive one. If you pass an exclusive
+        // one, then it might assert at you for being out of bounds. So we also
+        // take care of the case that the end point is outside the viewport
+        // manually.
```
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/ControlCore.cpp:2631]: it doesn't though 😢 `allowEndExclusive` plumbs through to:
https://github.com/microsoft/terminal/blob/8d67477a1a1b6d26669736057f4ef2f318400ac8/src/types/viewport.cpp#L211-L220
https://github.com/microsoft/terminal/blob/8d67477a1a1b6d26669736057f4ef2f318400ac8/src/types/viewport.cpp#L172-L175
https://github.com/microsoft/terminal/blob/8d67477a1a1b6d26669736057f4ef2f318400ac8/src/types/viewport.cpp#L116-L119

In this case, the right is exclusive, but not the bottom. So it fails the check in IsInBounds

File: src/cascadia/TerminalCore/terminalrenderdata.cpp [hunk 2]
```diff
@@ -199,7 +199,7 @@ til::CoordType Terminal::_ScrollToPoints(const til::point coordStart, const til:
         _NotifyScrollEvent();
     }
 
-    return _scrollOffset;
+    return _VisibleStartIndex();
```
Comment (tusharsnx) [inline src/cascadia/TerminalCore/terminalrenderdata.cpp:202]: Should we also update the comment, it currently reads:
```cpp
// Return Value:
// - The updated scroll offset
```

BTW, as someone new to this repo I have a doubt: what's the mutable viewport? and how does it differ from the viewport that `_scrollOffset` represents? 

Not the best place to ask questions, so I'm sorry 😅

File: src/cascadia/TerminalControl/ControlCore.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhecker) [reply src/cascadia/TerminalControl/ControlCore.cpp:2631]: Ugh... If you don't mind waiting for a PR from me, I can fix that once and for all for you.

File: src/cascadia/TerminalCore/terminalrenderdata.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalCore/terminalrenderdata.cpp:202]: No worries! I probably didn't do the best job with that particular abstraction, and it bites even me to this day.

image

(I should commit that somehere)

File: src/cascadia/TerminalControl/ControlCore.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhecker) [reply src/cascadia/TerminalControl/ControlCore.cpp:2631]: --> #17143

[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/ControlCore.cpp:2631]: As discussed - we're gonna move ahead with this _less-than-great_ solution for now and fix this in #17143 after 1.22 forks off

File: src/cascadia/TerminalCore/terminalrenderdata.cpp (hunk 2 above)
Comment (DHowett) [reply src/cascadia/TerminalCore/terminalrenderdata.cpp:202]: PUT THIS IN THE REPO SOMEWHERE

**Review by DHowett — APPROVED** [review]

**Review by lhecker — APPROVED** [review]

**Pull request merged**